maestro-ios-driver/src/main/kotlin/xcuitest/installer/LocalXCTestInstaller.kt

```diff
@@ -93,21 +93,21 @@ class LocalXCTestInstaller(
 
     override fun isChannelAlive(): Boolean {
         return XCRunnerCLIUtils.isAppAlive(UI_TEST_RUNNER_APP_BUNDLE_ID, deviceId) &&
-            subTreeOfRunnerApp().use { it.isSuccessful }
+            xcTestDriverStatusCheck().use { it.isSuccessful }
     }
 
     private fun ensureOpen(): Boolean {
         XCRunnerCLIUtils.ensureAppAlive(UI_TEST_RUNNER_APP_BUNDLE_ID, deviceId)
         return MaestroTimer.retryUntilTrue(10_000, 100) {
             try {
-                subTreeOfRunnerApp().use { it.isSuccessful }
-            } catch (ignore: ConnectException) {
+                xcTestDriverStatusCheck().use { it.isSuccessful }
+            } catch (ignore: IOException) {
```

Comment:
Because connect exception is just one case there could be many and we should only consider driver ready when it has no IOException